### DISCUSSION TITLE
Please recommend a most featured lisp implementation

### DISCUSSION THREAD

**Discussion by hww** [joaotavora/sly#523] [Q&A] [answered]
The manual tells

> The supported Lisp implementations, roughly ordered from the best-supported, are:
> 
> - CMU Common Lisp (CMUCL), 19d or newer
> - Steel Bank Common Lisp (SBCL), 1.0 or newer
> - Clozure Common Lisp (CCL), version 1.3 or newer
> - LispWorks, version 4.3 or newer
> - Allegro Common Lisp (ACL), version 6 or newer
> - CLISP, version 2.35 or newer
> - Armed Bear Common Lisp (ABCL)
> - Scieneer Common Lisp (SCL), version 1.2.7 or newer
> - Embedded Common Lisp (ECL)
> - ManKai Common Lisp (MKCL)
> - Clasp
>
> Most features work uniformly across implementations, but some are prone to variation. These include the precision of placing compiler-note annotations, XREF support, and fancy debugger commands (like “restart frame”).

Which of listed implementation has most advanced debugging features? Mostly interested in referencing to the source code location.

**Comment by joaotavora** [comment, marked as answer]
I recommend allegro common lisp or abcl, but that's because those were the
two i most used. Allegro has very nice tracing, function wrapping
facilities. sbcl is generally faster, had better code gen, has better
diagnostics.

**Reply by hww** [reply]
Thanks :)
